packages/core/src/core.ts

```diff
@@ -445,6 +463,14 @@ class ThresholdKey implements ITKey {
       for (let j = 0; j < tssIndex; j++) {
         _tssSharePub = _tssSharePub.add(tssCommitA1);
       }
+      if (accountIndex && accountIndex > 0) {
+        const nonce = this.computeNonce(accountIndex);
+        const derivedShare = tssShare.add(nonce).umod(ecCurve.n);
+        if (tssSharePub.getX().cmp(_tssSharePub.getX()) === 0 && tssSharePub.getY().cmp(_tssSharePub.getY()) === 0) {
+          return { tssIndex, tssShare: derivedShare };
+        }
+      }
+
       if (tssSharePub.getX().cmp(_tssSharePub.getX()) === 0 && tssSharePub.getY().cmp(_tssSharePub.getY()) === 0) {
```

Comment:
this and above if block can be merged together

```suggestion
     if (tssSharePub.getX().cmp(_tssSharePub.getX()) === 0 && tssSharePub.getY().cmp(_tssSharePub.getY()) === 0) {
     
      if (accountIndex && accountIndex > 0) {
        const nonce = this.computeNonce(accountIndex);
        const derivedShare = tssShare.add(nonce).umod(ecCurve.n);

          return { tssIndex, tssShare: derivedShare };

      }
      return { tssIndex, tssShare };
     }
```